torch_xla/csrc/xla_lower_util.cpp

```diff
@@ -749,4 +749,67 @@ xla::XlaOp BuildMaskedScatter(xla::XlaOp input, xla::XlaOp mask,
       scatter_dnums);
 }
 
+std::vector<xla::XlaOp> BuildAmpForachNonFiniteCheckAndUnscale(
```

Comment:
I think you mentioned `And my XlaOp could only handle the scaler one.` Could you point me to where is that limitation coming from?

Reply:
Ex. https://github.com/pytorch/xla/pull/2654/files#diff-5555c0238ce581790db5f184fb764328a269c974d5019e8d5b66bcedbc545fefR761, the result of `xla::AllReduce` is a scalar.

Reply:
Sorry could you give a bit more details? Why does `ReduceAll ` return type matters here?

Reply:
The result of `xla::AllReduce` is a scalar, which could not be used together with a `XlaOp` with shape `(1, )`.

Reply:
Good catch! Fixed.